### PR TITLE
skhd: submission of new port (v0.3.5)

### PR DESCRIPTION
**Pull request by TheKevJames**
#### Description
Create new port for the `skhd` project. [Link](https://github.com/koekeishiya/skhd).

This is my first Portfile submission, please let me know if I've made a mistake!

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix
- [x] submission

###### Tested on
macOS 10.15.5 19F101
Xcode 12.1 12A7403

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by ryandesign** [comment]
Because this is using `use_configure no`, you need to make arrangements to ensure that [the right compiler](https://trac.macports.org/wiki/UsingTheRightCompiler) and `-arch` flags are used. The makefile 1.0 portgroup might help you do that; read its source code to find out how.

**Comment by TheKevJames** [comment]
@ryandesign I think I may be missing something on this one. I read the [compiler docs page](https://trac.macports.org/wiki/UsingTheRightCompiler) and the [makefile 1.0 PortGroup source](https://github.com/macports/macports-ports/blob/master/_resources/port1.0/group/makefile-1.0.tcl), but it seems those are mostly methods of passing in command arguments such as:
```
build.args-append       CC=${configure.cc}
```
Unfortunately, the Makefile being referenced in this port looks like [this](https://github.com/koekeishiya/skhd/blob/master/makefile) -- most notably, the install block runs:
```
clang $^ $(BUILD_FLAGS) $(FRAMEWORKS) -o $@
```

Is there a method for correcting that hardcoded `clang` reference which I've missed in this PortGroup?

**Comment by ryandesign** [comment]
There is no one-size-fits-all answer to how to use the right compiler and flags with projects that use hand-made Makefiles. It all depends on how the author of the Makefile wrote it. In this case, if the developer hardcoded the compiler name "clang", that's not suitable and will need to be changed to `$(CC)`. Write a patchfile to make that and any other changes needed to get the Makefile to respect our variables and have the port use it with the `patchfiles` directive (e.g. you may need to make it use `$(CFLAGS)`, `$(CPPFLAGS)`, `$(CXXFLAGS)`, `$(LDFLAGS)` in the right places). You should send this patch to the developers so they can include it in the next version. If you can't develop such a patch, you can report the problem to the developers and ask them to fix it. Once the Makefile responds to the right variables, then the makefile 1.0 portgroup can help you supply the variables to the Makefile.

I see that you've disabled the port's build phase. Why? Ports should build in the build phase and install in the destroot phase.

**Comment by ryandesign** [comment]
> I see that you've disabled the port's build phase. Why? Ports should build in the build phase and install in the destroot phase.

Looking at the Makefile, I see that it builds with `-O0`, and when you install, it rebuilds with `-O2`. I agree we don't want that and as long as it works that way and since it's only compiling one thing I guess it's ok that it does it in the destroot phase and skips the build phase, but it's dumb that the Makefile is written that way and it should be fixed by the developer to work more normally.

**Comment by TheKevJames** [comment]
@ryandesign I've added a patchfile to address the issues you've mentioned. The port now: builds in the build phase and installs in the destroot phase, obeys `$CC` and `$CFLAGS`, and avoid building with `-g -O0` unnecessarily the first time. Would love a re-review!

File: sysutils/skhd/files/patch-Makefile.diff [hunk 1]
```diff
@@ -0,0 +1,25 @@
+--- makefile.orig	2020-12-03 18:18:40.000000000 -0800
++++ makefile	2020-12-03 18:18:48.000000000 -0800
+@@ -1,6 +1,6 @@
+ FRAMEWORKS     = -framework Carbon -framework CoreServices
+ BUILD_PATH     = ./bin
+-BUILD_FLAGS    = -std=c99 -Wall -g -O0
++CFLAGS         = -std=c99 -Wall -O3
```
Comment (reneeotten) [inline sysutils/skhd/files/patch-Makefile.diff:7]: just as in the other PR, this should likely become `CFLAGS         += -std=c99 -Wall` where the optimization, arch flags and such are automatically set by MacPorts.

**Comment by TheKevJames** [comment]
@reneeotten thanks for the tips! Updated with your recommendation and tests are now passing in CI.

**Comment by reneeotten** [comment]
> @reneeotten thanks for the tips! Updated with your recommendation and tests are now passing in CI.

okay, it still seems to build things both in the `build` phase and `destroot` phase. But it's so quick that I personally think it's fine.... and admittedly I don't know how to fix the Makefile so that it will not do that. Others might, and if someone points out how to do that, you could/should take that upstream and see if they are willing to fix it to follow the normal conventions (also with respect to their use of `BUILD_FLAGS` instead of `CFLAGS`.

**Pull request merged**